scheduler/plugin/plugin.go

```diff
@@ -366,13 +370,15 @@ func (pl *wasmPlugin) Reserve(ctx context.Context, state *framework.CycleState,
 
 // Unreserve implements the same method as documented on framework.ReservePlugin.
 func (pl *wasmPlugin) Unreserve(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) {
-	// Note: Unlike the below diagram, this is not a part of the scheduling
-	// cycle, rather the binding on error.
-	// https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/#extension-points
-
 	defer pl.pool.freeFromBinding(pod.UID) // the cycle is over, put it back into the pool.
 
-	// TODO: partially implemented for testing
+	params := &stack{pod: pod, nodeName: nodeName}
+	ctx = context.WithValue(ctx, stackKey{}, params)
+	if err := pl.pool.doWithSchedulingGuest(ctx, pod.UID, func(g *guest) {
+		g.unreserve(ctx)
+	}); err != nil {
+		klog.Error(err)
```

Comment:
This is the comment @sanposhiho left my PR.

Please use contextual logging, not directly use klog.Hoge.
klog has the contextual logging feature.
like https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go#L174

Reply:
a8309ad